Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1235,10 +1386,180 @@ CGImageRef CGJPEGImageCreateFromData(NSData* data) {
     return new CGJPEGDecoderImage(data);
 }
 
-bool CGContextIsPointInPath(CGContextRef c, bool eoFill, CGFloat x, CGFloat y) {
-    return c->Backing()->CGContextIsPointInPath(eoFill, x, y);
+////// CGBitmapContext
+struct __CGBitmapContext : __CGContext {
+    struct __CGBitmapContextImpl {
+        woc::unique_cf<CGImageRef> image;
+    };
+
+    __CGBitmapContextImpl _bitmapImpl;
+};
+
+static void __CGBitmapContextInit(CFTypeRef cf) {
+    // Delegate to parent constructor.
+    __CGContextInit(cf);
+
+    __CGBitmapContext* context = (__CGBitmapContext*)cf;
+    struct __CGBitmapContext* mutableContext = const_cast<struct __CGBitmapContext*>(context);
+    new (&mutableContext->_bitmapImpl) __CGBitmapContext::__CGBitmapContextImpl();
 }
 
-void CGContextDrawGlyphRun(CGContextRef ctx, const DWRITE_GLYPH_RUN* glyphRun) {
-    ctx->Backing()->CGContextDrawGlyphRun(glyphRun);
+static void __CGBitmapContextDeallocate(CFTypeRef cf) {
+    __CGBitmapContext* context = (__CGBitmapContext*)cf;
+    context->_bitmapImpl.~__CGBitmapContextImpl();
+
+    // Delegate to parent destructor.
+    __CGContextDeallocate(cf);
 }
+
+static CFTypeID __CGBitmapContextGetTypeID() {
+    static CFTypeID __kCGBitmapContextTypeID = _kCFRuntimeNotATypeID;
+    static const CFRuntimeClass __CGBitmapContextClass =
+        { 0, "CGBitmapContext", __CGBitmapContextInit, NULL, __CGBitmapContextDeallocate, NULL, NULL, NULL, __CGContextCopyDescription };
+
+    static dispatch_once_t initOnce = 0;
+    dispatch_once(&initOnce, ^{
+        __kCGBitmapContextTypeID = _CFRuntimeRegisterClass(&__CGBitmapContextClass);
+    });
+    return __kCGBitmapContextTypeID;
+}
+
+/**
+ @Status Caveat
+ @Notes Limited bitmap formats available. Decode, shouldInterpolate, intent parameters
+ and some byte orders ignored.
+ */
+CGContextRef CGBitmapContextCreate(void* data,
+                                   size_t width,
+                                   size_t height,
+                                   size_t bitsPerComponent,
+                                   size_t bytesPerRow,
+                                   CGColorSpaceRef colorSpace,
+                                   CGBitmapInfo bitmapInfo) {
+#if 0
```

Comment:
(This comment was on an old diff. Marking Resolved.)

---
In reply to: [82903055](https://github.com/Microsoft/WinObjC/pull/1126#discussion_r82903055) [](ancestors = 82903055)